Code/Tools/ProjectManager/Source/ProjectButtonWidget.h

```diff
@@ -37,62 +37,74 @@ namespace O3DE::ProjectManager
         explicit LabelButton(QWidget* parent = nullptr);
         ~LabelButton() = default;
 
-        void SetEnabled(bool enabled);
-        void SetOverlayText(const QString& text);
-        void SetLogUrl(const QUrl& url);
-
-        QLabel* GetOverlayLabel();
-        QProgressBar* GetProgressBar();
+        QLabel* GetMessageLabel();
+        QLabel* GetSubMessageLabel();
+        QLabel* GetWarningIcon();
+        QSpacerItem* GetWarningSpacer();
+        QLabel* GetBuildingAnimationLabel();
         QPushButton* GetOpenEditorButton();
         QPushButton* GetActionButton();
-        QLabel* GetWarningLabel();
-        QLabel* GetWarningIcon();
-        QLayout* GetBuildOverlayLayout();
-
-    signals:
-        void triggered(QMouseEvent* event);
+        QPushButton* GetActionCancelButton();
+        QPushButton* GetShowLogsButton();
+        QLabel* GetDarkenOverlay();
 
     public slots:
         void mousePressEvent(QMouseEvent* event) override;
-        void OnLinkActivated(const QString& link);
+
+    signals:
+        void triggered(QMouseEvent* event);
 
     private:
-        QVBoxLayout* m_buildOverlayLayout = nullptr;
-        QLabel* m_overlayLabel = nullptr;
-        QProgressBar* m_progressBar = nullptr;
-        QPushButton* m_openEditorButton = nullptr;
-        QPushButton* m_actionButton = nullptr;
-        QLabel* m_warningText = nullptr;
+        QVBoxLayout* m_projectOverlayLayout = nullptr;
+
+        QLabel* m_darkenOverlay = nullptr;
+
+        QLabel* m_messageLabel = nullptr;
+        QLabel* m_subMessageLabel = nullptr;
+
         QLabel* m_warningIcon = nullptr;
+        QSpacerItem* m_warningSpacer = nullptr;
 
-        QUrl m_logUrl;
-        bool m_enabled = true;
+        QLabel* m_buildingAnimation = nullptr;
+
+        QPushButton* m_openEditorButton = nullptr;
+        QPushButton* m_actionButton = nullptr;
+        QPushButton* m_actionCancelButton = nullptr;
+        QPushButton* m_showLogsButton = nullptr;
     };
 
     class ProjectButton
         : public QFrame
     {
-        Q_OBJECT // AUTOMOC
+        Q_OBJECT
 
     public:
+        enum class ProjectState
+        {
+            ReadyToLaunch = 0,
+            Launching,
+            NeedsToBuild,
+            Building,
+            BuildFailed
+        };
```

Comment:
Are we duplicating state here? I noticed projects have m_needsBuild and m_buildFailed, and I'm wondering if we shouldn't just use the state on the `m_projectInfo`

Reply:
Unfortunately there multiple copies of projectInfo floating around so the state is already copied. Those variables are also inconsistently used on projectInfo so there is now guarantee that they are in the correct state when a projectInfo gets passed to the ProjectButton on creation. The potential project states vs project button states also do have some parallels but there are other state only unique to the button such as, Building, Launching, and more potential future state. 

For now I can change the name from `ProjectState` to `ProjectButtonState` to make it more clear this is a distinct set of states.